node/process_test.ts

```diff
@@ -326,6 +326,66 @@ Deno.test({
   },
 });
 
+Deno.test({
+  name: "process.stdin readable with a TTY",
+  ignore: !Deno.isatty(Deno.stdin.rid),
+  async fn() {
+    const promise = deferred();
+    const expected = ["foo", "bar", null, "end"];
+    const data: (string | null)[] = [];
+
+    process.stdin.setEncoding("utf8");
+    process.stdin.on("readable", () => {
+      data.push(process.stdin.read());
+    });
+    process.stdin.on("end", () => {
+      data.push("end");
+    });
+
+    process.stdin.push("foo");
+    process.nextTick(() => {
+      process.stdin.push("bar");
+      process.nextTick(() => {
+        process.stdin.push(null);
+        promise.resolve();
+      });
+    });
+
+    await promise;
+    assertEquals(process.stdin.readableHighWaterMark, 0);
+    assertEquals(data, expected);
+  },
+});
+
+Deno.test({
+  name: "process.stdin readable with piping a file",
+  async fn() {
+    const expected = ["65536", "foo", "bar", "null", "end"];
```

Comment:
In node.js, the first number becomes `16384`. Is this difference intentional?

Reply:
My confirmation was lacking.
As I mentioned in the [issue](https://github.com/denoland/deno_std/issues/2974#issuecomment-1382845551), the highWaterMark is `65536` with piping a file, But when running the code in child_process with piping stdin, the highWaterMark is a different value.

- tty(tty.ReadStream): 0
- piping files(fs.ReadStream): 65536
- piping stdin(Stream?): 16384

I'll look into it further.
Thank you for pointing this out.